tools.py

```diff
@@ -323,6 +324,24 @@ async def generate_pac_crate(
 
     async with CRATE_GEN_SEM:
         with tempfile.TemporaryDirectory() as tmpd:
+            # Patch MCU SVD if needed
+            if patch != None:
+                with open(f"{patch}") as fp:
+                    patch_content = fp.read()
+                tmp_patch = f"{tmpd}/{patch.name}"
+                with open(f"{tmp_patch}", 'w') as fp:
+                    fp.write(patch_content.format(svd_descr.path))
+                pret = await asyncio.create_subprocess_exec(
+                    *[
+                        "svdtools",
+                        "patch",
+                        f"{tmp_patch}",
+                    ],
+                )
+                await pret.wait()
+                assert pret.returncode == 0
+                shutil.move(svd_descr.path.with_suffix(".svd.patched"), f"{tmpd}")
+                svd_descr.path = f"{tmpd}/{svd_descr.path.with_suffix('.svd.patched').name}"
```

Comment:
Would suggest to wrap `tmpd` into a `pathlib.Path` and use either `.joinpath()` or `/` overload as usage of `"../.."` delimiters is platform dependent:

```py
tmpd_path = pathlib.Path(tmpd)
# ...
svd_descr.path = tmpd / f"{svd_descr.path.with_suffix('.svd.patched').name}"
```